hngprojects/hng_boilerplate_nestjs
# feat: get-products

## What does the PR do?

 Developed an API endpoint to list products with pagination support.

- [x] get all product.
- [x] products pagination

How should this be manually tested?


1. Make a Get request to the following endpoint 
   1. `\api\v1\products` 
    2. `\api\v1\products\?limit=10&page=1` 
2. The following payload must be put in place:

```json
Authorization: Bearer <access_token>
```
Successful Response

On a successful product list, the API should return a 200 get status code.

```json
{
  "success": true,
  "message": "Product retrieved successfully",
  "products": [
    {
      "name": "Product 1",
      "price": 29.99,
      "category": "Electronics"
    },
    {
      "name": "Product 2",
      "price": 19.99,
      "category": "Books"
    },
    ...
  ],
  "pagination": {
    "totalItems": 100,
    "totalPages": 10,
    "currentPage": 1
  },
"status_code": 200
}
```
![success](https://github.com/user-attachments/assets/a2ec80e7-a413-4633-afae-0cc7a364194c)


Authentication Error:

When the current user is not properly authenticated to change the organization detail:

```json
{
  "message": "User is currently unauthorized, kindly authenticate to continue",
  "status_code": 401
}
```
![error](https://github.com/user-attachments/assets/2daeaf6a-b246-4744-8d19-e6c785438017)

Task


- [x] Make a get request for product.
- [x] Make a pagination query
Reference Issue: [Link to Issue](https://github.com/hngprojects/hng_boilerplate_nestjs/issues/97)
